src/SolverInterface/constraints.jl

```diff
@@ -0,0 +1,134 @@
+
+
+
+
+
+"""
+    addconstraint!(m::AbstractMathProgModel, b, a_constridx, a_varidx, a_coef, Q_constridx, Q_vari, Q_varj, Q_coef, S::AbstractSet)::QuadraticConstraintReference{typeof(S)}
+
+Add the quadratic-in-set constraint
+```math
+Ax + b + q(x) \\in S
+```
+where ``A`` is a sparse matrix specified in triplet form by
+`a_constridx`, `a_varidx`, and `a_coef`; ``b`` is a vector;
+``q(x)`` is a vector with component ``(q(x))_k`` defined to be ``\\frac{1}{2}x^TQ_kx``
+where the symmetric matrix ``Q_k`` is defined by the triplets in `Q_vari`, `Q_varj`,
+`Q_coef` for which `Q_constridx` equals `k`; and the set ``S`` is defined by `S`.
+
+Duplicate indices in either ``A`` or  a ``Q`` matrix are accepted and will be summed together. Off-diagonal entries of ``Q`` will be mirrored, so either the upper triangular or lower triangular entries of ``Q`` should be provided. If entries for both ``(i,j)`` and ``(j,i)`` are provided, these are considered duplicate terms. `a_varidx`, `Q_vari`, `Q_varj` should be collections of `VariableReference` objects.
```

Comment:
It is unclear what "mirrored" means. Does `2xy` corresponds to `xy + yx` or `2xy + 2yx` ?